### ISSUE TITLE
Debugging check

### ISSUE THREAD

**Issue by vomikan** [moneymanagerex/moneymanagerex#3721]
![image](https://user-images.githubusercontent.com/6836805/126993143-9da66c39-3e8e-4812-8fee-d2b68e66f30f.png)

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/commit/db77afc7f62118d7e1120c2c1812b7b7ecaa3664

Possible fix:
![image](https://user-images.githubusercontent.com/6836805/126993774-a657a0ec-6b35-4c54-ab41-1c4f3edc4249.png)

**Comment by lonemadmax** [comment]
If you force the English locale you, well, get the English locale, which is not what you want?

I'm not sure how the test works. Is that assert triggered by the call to FromCDouble? That doesn't seem to make sense. FromCDouble is explicitly used to get the C format (which I guess is what we really want here if we are writing json), not whatever current locale is.

I've found https://github.com/wxWidgets/wxWidgets/pull/2330, does that by any chance fix the test?

**Comment by vomikan** [comment]
Why is the American locale used during initialization? Because it is predictable. Next, we will convert the numbers and dates to the type that the user has chosen. If you use the default locale, then an element of unpredictability appears.

**Comment by lonemadmax** [comment]
Not really. If you use the default locale, you have what the user has set up for their system. If you use the US locale, you get (for example) the so much hated outside the US MM/DD/YYYY date format in the elements you don't control.

I say that assert is wrong if it comes from the call to FromCDouble. There's no unpredictability there and that function is meant to return the same string whatever the locale. Now, the implementation may be to temporarily change to C locale, format the string and then change back to whatever the locale was. If that's the case, the check is done after the last step and the problem commented in the mentioned wxWidgets patch is there, the assert triggers. Not because you've done something wrong, but because a bug in wxWidgets.

If you are building your own wxWidgets, it should be easy to check if it still happens with that patch applied.

If that's really the root cause, and given that we don't bundle wxWidgets in mmex, https://github.com/moneymanagerex/moneymanagerex/commit/db77afc7f62118d7e1120c2c1812b7b7ecaa3664 can be reverted until wxWidgets releases a version with the fix and mmex documents it as required.
